src/client/cli/cmd/register.cpp

```diff
@@ -62,26 +64,52 @@ QString cmd::Register::description() const
 
 mp::ParseCode cmd::Register::parse_args(mp::ArgParser* parser)
 {
-    parser->addPositionalArgument("passphrase", "Trusted passphrase to send to Multipass service", "<passphrase>");
+    parser->addPositionalArgument("passphrase",
+                                  "Trusted passphrase to send to the Multipass service. If omitted, a prompt will be "
+                                  "displayed for entering the passphrase.",
+                                  "<passphrase>");
 
     auto status = parser->commandParse(this);
     if (status != ParseCode::Ok)
         return status;
 
-    // TODO: Add the echoless passphrase handling here
-    if (parser->positionalArguments().count() == 0)
-    {
-        cerr << "No passphrase given\n";
-        return ParseCode::CommandLineError;
-    }
-
     if (parser->positionalArguments().count() > 1)
     {
         cerr << "Too many passphrases given\n";
         return ParseCode::CommandLineError;
     }
 
-    request.set_passphrase(parser->positionalArguments().first().toStdString());
+    if (parser->positionalArguments().empty())
+    {
+        if (!term->is_live())
+        {
+            cerr << "The terminal is not live: The passphrase argument is required\n";
+            return ParseCode::CommandLineError;
+        }
+
```

Comment:
```suggestion
```

User could have piped it in, for whatever reason?

Reply:
Yep, done.